nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/store/LocalRequestTraceStore.java

```diff
@@ -98,20 +93,20 @@ public RequestTrace addTrace(RequestTrace trace, RequestTrace traceToRemove) {
 
     @Override
     public void setSize(int maxSize) {
-        // Set the store to be one bigger than the specified value. Allows the store to transparently select which item to remove
-        this.store = new BoundedTreeSet<>(maxSize + 1);
+        while (store.size() > maxSize) {
+            store.remove(strategy.getTraceForRemoval(getTraces(), maxSize, null));
+        }
```

Comment:
Again changing size updates the store if needed (same as above in other strategy)